--- sdk/vendor-disasm/v2_0_2_0/libdnvqe/hi_audio_dl_work.S	2026-03-29 07:34:59.078640160 +0300
+++ sdk/vendor-disasm/v2_0_2_1/libdnvqe/hi_audio_dl_work.S	2026-03-29 07:34:59.971975931 +0300
@@ -2,53 +2,54 @@
 hi_audio_dl_work.o:     file format elf32-littlearm
 
 SYMBOL TABLE:
-00000000 l       .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .LC0
-0000000f l       .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .LC1
-0000001e l       .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .LC2
-0000002d l       .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .LC3
-0000003b l       .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .LC4
-00000070 l       .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .LC5
-0000002c l       .rodata.DL_HI_Audio_Dlsym.str1.1	00000000 .LC7
-00000000 l       .rodata.DL_HI_Audio_Dlsym.str1.1	00000000 .LC6
-00000040 l       .rodata.DL_HI_Audio_Dlsym.str1.1	00000000 .LC8
-00000000 l    d  .rodata.__FUNCTION__.6278	00000000 .rodata.__FUNCTION__.6278
-00000000 l    d  .rodata.__FUNCTION__.6282	00000000 .rodata.__FUNCTION__.6282
+00000000 l       .rodata.DL_hi_audio_dlopen.str1.1	00000000 .LC0
+0000000f l       .rodata.DL_hi_audio_dlopen.str1.1	00000000 .LC1
+0000001e l       .rodata.DL_hi_audio_dlopen.str1.1	00000000 .LC2
+0000002d l       .rodata.DL_hi_audio_dlopen.str1.1	00000000 .LC3
+0000003b l       .rodata.DL_hi_audio_dlopen.str1.1	00000000 .LC4
+0000008b l       .rodata.DL_hi_audio_dlopen.str1.1	00000000 .LC5
+00000000 l       .rodata.DL_hi_audio_dlsym.str1.1	00000000 .LC6
+00000030 l       .rodata.DL_hi_audio_dlsym.str1.1	00000000 .LC7
+0000005c l       .rodata.DL_hi_audio_dlsym.str1.1	00000000 .LC8
+00000000 l    d  .rodata.__FUNCTION__.6323	00000000 .rodata.__FUNCTION__.6323
+00000000 l    d  .rodata.__FUNCTION__.6330	00000000 .rodata.__FUNCTION__.6330
+00000000 l    d  .rodata.__FUNCTION__.6334	00000000 .rodata.__FUNCTION__.6334
 00000000 l    d  .text	00000000 .text
 00000000 l    d  .data	00000000 .data
 00000000 l    d  .bss	00000000 .bss
-00000000 l    d  .text.DL_HI_Audio_Dlopen	00000000 .text.DL_HI_Audio_Dlopen
-00000000 l    d  .text.DL_HI_Audio_Dlsym	00000000 .text.DL_HI_Audio_Dlsym
-00000000 l    d  .text.DL_HI_Audio_Dlclose	00000000 .text.DL_HI_Audio_Dlclose
-00000000 l    d  .rodata.DL_HI_Audio_Dlopen.str1.1	00000000 .rodata.DL_HI_Audio_Dlopen.str1.1
-00000000 l    d  .rodata.DL_HI_Audio_Dlsym.str1.1	00000000 .rodata.DL_HI_Audio_Dlsym.str1.1
+00000000 l    d  .text.DL_hi_audio_dlopen	00000000 .text.DL_hi_audio_dlopen
+00000000 l    d  .text.DL_hi_audio_dlsym	00000000 .text.DL_hi_audio_dlsym
+00000000 l    d  .text.DL_hi_audio_dlclose	00000000 .text.DL_hi_audio_dlclose
+00000000 l    d  .rodata.DL_hi_audio_dlopen.str1.1	00000000 .rodata.DL_hi_audio_dlopen.str1.1
+00000000 l    d  .rodata.DL_hi_audio_dlsym.str1.1	00000000 .rodata.DL_hi_audio_dlsym.str1.1
 00000000 l    d  .comment	00000000 .comment
 00000000 l    d  .note.GNU-stack	00000000 .note.GNU-stack
 00000000 l    d  .ARM.attributes	00000000 .ARM.attributes
-00000000 g     F .text.DL_HI_Audio_Dlopen	00000100 DL_HI_Audio_Dlopen
+00000000 g     F .text.DL_hi_audio_dlopen	00000120 DL_hi_audio_dlopen
 00000000         *UND*	00000000 dlopen
 00000000         *UND*	00000000 strcmp
 00000000         *UND*	00000000 fprintf
 00000000         *UND*	00000000 _GLOBAL_OFFSET_TABLE_
 00000000         *UND*	00000000 stderr
-00000000 g     F .text.DL_HI_Audio_Dlsym	000000e0 DL_HI_Audio_Dlsym
+00000000 g     F .text.DL_hi_audio_dlsym	000000dc DL_hi_audio_dlsym
 00000000         *UND*	00000000 dlsym
 00000000         *UND*	00000000 dlerror
-00000000 g     F .text.DL_HI_Audio_Dlclose	00000070 DL_HI_Audio_Dlclose
+00000000 g     F .text.DL_hi_audio_dlclose	0000005c DL_hi_audio_dlclose
 00000000         *UND*	00000000 dlclose
 
 
 
-Disassembly of section .text.DL_HI_Audio_Dlopen:
+Disassembly of section .text.DL_hi_audio_dlopen:
 
-00000000 <DL_HI_Audio_Dlopen>:
-   0:	e92d41f0 	push	{r4, r5, r6, r7, r8, lr}
+00000000 <DL_hi_audio_dlopen>:
+   0:	e92d40f7 	push	{r0, r1, r2, r4, r5, r6, r7, lr}
    4:	e3510000 	cmp	r1, #0
    8:	13500000 	cmpne	r0, #0
-   c:	e59f70cc 	ldr	r7, [pc, #204]	; e0 <DL_HI_Audio_Dlopen+0xe0>
+   c:	e59f70e4 	ldr	r7, [pc, #228]	; f8 <DL_hi_audio_dlopen+0xf8>
   10:	03a05001 	moveq	r5, #1
   14:	13a05000 	movne	r5, #0
   18:	e08f7007 	add	r7, pc, r7
-  1c:	0a000027 	beq	c0 <DL_HI_Audio_Dlopen+0xc0>
+  1c:	0a00002b 	beq	d0 <DL_hi_audio_dlopen+0xd0>
   20:	e5805000 	str	r5, [r0]
   24:	e1a04001 	mov	r4, r1
   28:	e1a06000 	mov	r6, r0
@@ -58,176 +59,178 @@
 			34: R_ARM_CALL	dlopen
   38:	e5860000 	str	r0, [r6]
   3c:	e3500000 	cmp	r0, #0
-  40:	1a000024 	bne	d8 <DL_HI_Audio_Dlopen+0xd8>
-  44:	e59f1098 	ldr	r1, [pc, #152]	; e4 <DL_HI_Audio_Dlopen+0xe4>
-  48:	e1a00004 	mov	r0, r4
-  4c:	e08f1001 	add	r1, pc, r1
-  50:	ebfffffe 	bl	0 <strcmp>
-			50: R_ARM_CALL	strcmp
-  54:	e59f308c 	ldr	r3, [pc, #140]	; e8 <DL_HI_Audio_Dlopen+0xe8>
-  58:	e3500000 	cmp	r0, #0
-  5c:	e7975003 	ldr	r5, [r7, r3]
-  60:	0a000011 	beq	ac <DL_HI_Audio_Dlopen+0xac>
-  64:	e59f1080 	ldr	r1, [pc, #128]	; ec <DL_HI_Audio_Dlopen+0xec>
-  68:	e1a00004 	mov	r0, r4
-  6c:	e08f1001 	add	r1, pc, r1
-  70:	ebfffffe 	bl	0 <strcmp>
-			70: R_ARM_CALL	strcmp
-  74:	e3500000 	cmp	r0, #0
-  78:	0a00000b 	beq	ac <DL_HI_Audio_Dlopen+0xac>
-  7c:	e59f106c 	ldr	r1, [pc, #108]	; f0 <DL_HI_Audio_Dlopen+0xf0>
-  80:	e1a00004 	mov	r0, r4
-  84:	e08f1001 	add	r1, pc, r1
-  88:	ebfffffe 	bl	0 <strcmp>
-			88: R_ARM_CALL	strcmp
-  8c:	e3500000 	cmp	r0, #0
-  90:	0a000005 	beq	ac <DL_HI_Audio_Dlopen+0xac>
-  94:	e59f1058 	ldr	r1, [pc, #88]	; f4 <DL_HI_Audio_Dlopen+0xf4>
-  98:	e1a00004 	mov	r0, r4
-  9c:	e08f1001 	add	r1, pc, r1
-  a0:	ebfffffe 	bl	0 <strcmp>
-			a0: R_ARM_CALL	strcmp
-  a4:	e3500000 	cmp	r0, #0
-  a8:	1a000006 	bne	c8 <DL_HI_Audio_Dlopen+0xc8>
-  ac:	e59f1044 	ldr	r1, [pc, #68]	; f8 <DL_HI_Audio_Dlopen+0xf8>
-  b0:	e1a02004 	mov	r2, r4
-  b4:	e08f1001 	add	r1, pc, r1
-  b8:	e5950000 	ldr	r0, [r5]
-  bc:	ebfffffe 	bl	0 <fprintf>
-			bc: R_ARM_CALL	fprintf
-  c0:	e3e00000 	mvn	r0, #0
-  c4:	e8bd81f0 	pop	{r4, r5, r6, r7, r8, pc}
-  c8:	e59f102c 	ldr	r1, [pc, #44]	; fc <DL_HI_Audio_Dlopen+0xfc>
-  cc:	e1a02004 	mov	r2, r4
-  d0:	e08f1001 	add	r1, pc, r1
-  d4:	eafffff7 	b	b8 <DL_HI_Audio_Dlopen+0xb8>
-  d8:	e1a00005 	mov	r0, r5
-  dc:	e8bd81f0 	pop	{r4, r5, r6, r7, r8, pc}
-  e0:	000000c0 	.word	0x000000c0
-			e0: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
-  e4:	00000090 	.word	0x00000090
-			e4: R_ARM_REL32	.LC0
-  e8:	00000000 	.word	0x00000000
-			e8: R_ARM_GOT32	stderr
-  ec:	00000078 	.word	0x00000078
-			ec: R_ARM_REL32	.LC1
-  f0:	00000064 	.word	0x00000064
-			f0: R_ARM_REL32	.LC2
-  f4:	00000050 	.word	0x00000050
-			f4: R_ARM_REL32	.LC3
-  f8:	0000003c 	.word	0x0000003c
-			f8: R_ARM_REL32	.LC4
-  fc:	00000024 	.word	0x00000024
-			fc: R_ARM_REL32	.LC5
+  40:	11a00005 	movne	r0, r5
+  44:	1a000022 	bne	d4 <DL_hi_audio_dlopen+0xd4>
+  48:	e59f10ac 	ldr	r1, [pc, #172]	; fc <DL_hi_audio_dlopen+0xfc>
+  4c:	e1a00004 	mov	r0, r4
+  50:	e08f1001 	add	r1, pc, r1
+  54:	ebfffffe 	bl	0 <strcmp>
+			54: R_ARM_CALL	strcmp
+  58:	e59f30a0 	ldr	r3, [pc, #160]	; 100 <DL_hi_audio_dlopen+0x100>
+  5c:	e3500000 	cmp	r0, #0
+  60:	e7975003 	ldr	r5, [r7, r3]
+  64:	0a000011 	beq	b0 <DL_hi_audio_dlopen+0xb0>
+  68:	e59f1094 	ldr	r1, [pc, #148]	; 104 <DL_hi_audio_dlopen+0x104>
+  6c:	e1a00004 	mov	r0, r4
+  70:	e08f1001 	add	r1, pc, r1
+  74:	ebfffffe 	bl	0 <strcmp>
+			74: R_ARM_CALL	strcmp
+  78:	e3500000 	cmp	r0, #0
+  7c:	0a00000b 	beq	b0 <DL_hi_audio_dlopen+0xb0>
+  80:	e59f1080 	ldr	r1, [pc, #128]	; 108 <DL_hi_audio_dlopen+0x108>
+  84:	e1a00004 	mov	r0, r4
+  88:	e08f1001 	add	r1, pc, r1
+  8c:	ebfffffe 	bl	0 <strcmp>
+			8c: R_ARM_CALL	strcmp
+  90:	e3500000 	cmp	r0, #0
+  94:	0a000005 	beq	b0 <DL_hi_audio_dlopen+0xb0>
+  98:	e59f106c 	ldr	r1, [pc, #108]	; 10c <DL_hi_audio_dlopen+0x10c>
+  9c:	e1a00004 	mov	r0, r4
+  a0:	e08f1001 	add	r1, pc, r1
+  a4:	ebfffffe 	bl	0 <strcmp>
+			a4: R_ARM_CALL	strcmp
+  a8:	e3500000 	cmp	r0, #0
+  ac:	1a00000a 	bne	dc <DL_hi_audio_dlopen+0xdc>
+  b0:	e59f2058 	ldr	r2, [pc, #88]	; 110 <DL_hi_audio_dlopen+0x110>
+  b4:	e3a03027 	mov	r3, #39	; 0x27
+  b8:	e59f1054 	ldr	r1, [pc, #84]	; 114 <DL_hi_audio_dlopen+0x114>
+  bc:	e08f2002 	add	r2, pc, r2
+  c0:	e08f1001 	add	r1, pc, r1
+  c4:	e58d4000 	str	r4, [sp]
+  c8:	e5950000 	ldr	r0, [r5]
+  cc:	ebfffffe 	bl	0 <fprintf>
+			cc: R_ARM_CALL	fprintf
+  d0:	e3e00000 	mvn	r0, #0
+  d4:	e28dd00c 	add	sp, sp, #12
+  d8:	e8bd80f0 	pop	{r4, r5, r6, r7, pc}
+  dc:	e59f2034 	ldr	r2, [pc, #52]	; 118 <DL_hi_audio_dlopen+0x118>
+  e0:	e3a0302d 	mov	r3, #45	; 0x2d
+  e4:	e59f1030 	ldr	r1, [pc, #48]	; 11c <DL_hi_audio_dlopen+0x11c>
+  e8:	e58d4000 	str	r4, [sp]
+  ec:	e08f2002 	add	r2, pc, r2
+  f0:	e08f1001 	add	r1, pc, r1
+  f4:	eafffff3 	b	c8 <DL_hi_audio_dlopen+0xc8>
+  f8:	000000d8 	.word	0x000000d8
+			f8: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
+  fc:	000000a4 	.word	0x000000a4
+			fc: R_ARM_REL32	.LC0
+ 100:	00000000 	.word	0x00000000
+			100: R_ARM_GOT32	stderr
+ 104:	0000008c 	.word	0x0000008c
+			104: R_ARM_REL32	.LC1
+ 108:	00000078 	.word	0x00000078
+			108: R_ARM_REL32	.LC2
+ 10c:	00000064 	.word	0x00000064
+			10c: R_ARM_REL32	.LC3
+ 110:	0000004c 	.word	0x0000004c
+			110: R_ARM_REL32	.rodata.__FUNCTION__.6323
+ 114:	0000004c 	.word	0x0000004c
+			114: R_ARM_REL32	.LC4
+ 118:	00000024 	.word	0x00000024
+			118: R_ARM_REL32	.rodata.__FUNCTION__.6323
+ 11c:	00000024 	.word	0x00000024
+			11c: R_ARM_REL32	.LC5
 
-Disassembly of section .text.DL_HI_Audio_Dlsym:
+Disassembly of section .text.DL_hi_audio_dlsym:
 
-00000000 <DL_HI_Audio_Dlsym>:
+00000000 <DL_hi_audio_dlsym>:
    0:	e92d40f7 	push	{r0, r1, r2, r4, r5, r6, r7, lr}
    4:	e1a07000 	mov	r7, r0
-   8:	e59f50b0 	ldr	r5, [pc, #176]	; c0 <DL_HI_Audio_Dlsym+0xc0>
+   8:	e59f50b0 	ldr	r5, [pc, #176]	; c0 <DL_hi_audio_dlsym+0xc0>
    c:	e2510000 	subs	r0, r1, #0
   10:	e08f5005 	add	r5, pc, r5
-  14:	1a00000e 	bne	54 <DL_HI_Audio_Dlsym+0x54>
-  18:	e59f30a4 	ldr	r3, [pc, #164]	; c4 <DL_HI_Audio_Dlsym+0xc4>
-  1c:	e59f20a4 	ldr	r2, [pc, #164]	; c8 <DL_HI_Audio_Dlsym+0xc8>
+  14:	1a00000b 	bne	48 <DL_hi_audio_dlsym+0x48>
+  18:	e59f30a4 	ldr	r3, [pc, #164]	; c4 <DL_hi_audio_dlsym+0xc4>
+  1c:	e59f20a4 	ldr	r2, [pc, #164]	; c8 <DL_hi_audio_dlsym+0xc8>
   20:	e7950003 	ldr	r0, [r5, r3]
-  24:	e59f30a0 	ldr	r3, [pc, #160]	; cc <DL_HI_Audio_Dlsym+0xcc>
-  28:	e59f10a0 	ldr	r1, [pc, #160]	; d0 <DL_HI_Audio_Dlsym+0xd0>
-  2c:	e08f3003 	add	r3, pc, r3
-  30:	e58d3000 	str	r3, [sp]
-  34:	e3a0303b 	mov	r3, #59	; 0x3b
-  38:	e08f2002 	add	r2, pc, r2
-  3c:	e08f1001 	add	r1, pc, r1
-  40:	e5900000 	ldr	r0, [r0]
-  44:	ebfffffe 	bl	0 <fprintf>
-			44: R_ARM_CALL	fprintf
-  48:	e3e00000 	mvn	r0, #0
-  4c:	e28dd00c 	add	sp, sp, #12
-  50:	e8bd80f0 	pop	{r4, r5, r6, r7, pc}
-  54:	e3a04000 	mov	r4, #0
-  58:	e1a01002 	mov	r1, r2
-  5c:	e5874000 	str	r4, [r7]
-  60:	e1a06002 	mov	r6, r2
-  64:	ebfffffe 	bl	0 <dlsym>
-			64: R_ARM_CALL	dlsym
-  68:	e1500004 	cmp	r0, r4
-  6c:	e5870000 	str	r0, [r7]
-  70:	11a00004 	movne	r0, r4
-  74:	1afffff4 	bne	4c <DL_HI_Audio_Dlsym+0x4c>
-  78:	e59f3044 	ldr	r3, [pc, #68]	; c4 <DL_HI_Audio_Dlsym+0xc4>
-  7c:	e59f2050 	ldr	r2, [pc, #80]	; d4 <DL_HI_Audio_Dlsym+0xd4>
-  80:	e7954003 	ldr	r4, [r5, r3]
-  84:	e3a03049 	mov	r3, #73	; 0x49
-  88:	e59f1048 	ldr	r1, [pc, #72]	; d8 <DL_HI_Audio_Dlsym+0xd8>
-  8c:	e58d6000 	str	r6, [sp]
-  90:	e08f2002 	add	r2, pc, r2
-  94:	e08f1001 	add	r1, pc, r1
-  98:	e5940000 	ldr	r0, [r4]
-  9c:	ebfffffe 	bl	0 <fprintf>
-			9c: R_ARM_CALL	fprintf
-  a0:	e5944000 	ldr	r4, [r4]
-  a4:	ebfffffe 	bl	0 <dlerror>
-			a4: R_ARM_CALL	dlerror
-  a8:	e59f102c 	ldr	r1, [pc, #44]	; dc <DL_HI_Audio_Dlsym+0xdc>
-  ac:	e1a02000 	mov	r2, r0
-  b0:	e1a00004 	mov	r0, r4
-  b4:	e08f1001 	add	r1, pc, r1
+  24:	e3a0303c 	mov	r3, #60	; 0x3c
+  28:	e59f109c 	ldr	r1, [pc, #156]	; cc <DL_hi_audio_dlsym+0xcc>
+  2c:	e08f2002 	add	r2, pc, r2
+  30:	e08f1001 	add	r1, pc, r1
+  34:	e5900000 	ldr	r0, [r0]
+  38:	ebfffffe 	bl	0 <fprintf>
+			38: R_ARM_CALL	fprintf
+  3c:	e3e00000 	mvn	r0, #0
+  40:	e28dd00c 	add	sp, sp, #12
+  44:	e8bd80f0 	pop	{r4, r5, r6, r7, pc}
+  48:	e3a04000 	mov	r4, #0
+  4c:	e1a01002 	mov	r1, r2
+  50:	e5874000 	str	r4, [r7]
+  54:	e1a06002 	mov	r6, r2
+  58:	ebfffffe 	bl	0 <dlsym>
+			58: R_ARM_CALL	dlsym
+  5c:	e1500004 	cmp	r0, r4
+  60:	e5870000 	str	r0, [r7]
+  64:	11a00004 	movne	r0, r4
+  68:	1afffff4 	bne	40 <DL_hi_audio_dlsym+0x40>
+  6c:	e59f3050 	ldr	r3, [pc, #80]	; c4 <DL_hi_audio_dlsym+0xc4>
+  70:	e59f4058 	ldr	r4, [pc, #88]	; d0 <DL_hi_audio_dlsym+0xd0>
+  74:	e7955003 	ldr	r5, [r5, r3]
+  78:	e3a03049 	mov	r3, #73	; 0x49
+  7c:	e59f1050 	ldr	r1, [pc, #80]	; d4 <DL_hi_audio_dlsym+0xd4>
+  80:	e08f4004 	add	r4, pc, r4
+  84:	e58d6000 	str	r6, [sp]
+  88:	e1a02004 	mov	r2, r4
+  8c:	e08f1001 	add	r1, pc, r1
+  90:	e5950000 	ldr	r0, [r5]
+  94:	ebfffffe 	bl	0 <fprintf>
+			94: R_ARM_CALL	fprintf
+  98:	e5955000 	ldr	r5, [r5]
+  9c:	ebfffffe 	bl	0 <dlerror>
+			9c: R_ARM_CALL	dlerror
+  a0:	e59f1030 	ldr	r1, [pc, #48]	; d8 <DL_hi_audio_dlsym+0xd8>
+  a4:	e3a0304b 	mov	r3, #75	; 0x4b
+  a8:	e58d0000 	str	r0, [sp]
+  ac:	e1a02004 	mov	r2, r4
+  b0:	e08f1001 	add	r1, pc, r1
+  b4:	e1a00005 	mov	r0, r5
   b8:	ebfffffe 	bl	0 <fprintf>
 			b8: R_ARM_CALL	fprintf
-  bc:	eaffffe1 	b	48 <DL_HI_Audio_Dlsym+0x48>
+  bc:	eaffffde 	b	3c <DL_hi_audio_dlsym+0x3c>
   c0:	000000a8 	.word	0x000000a8
 			c0: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
   c4:	00000000 	.word	0x00000000
 			c4: R_ARM_GOT32	stderr
-  c8:	00000088 	.word	0x00000088
-			c8: R_ARM_REL32	.rodata.__FUNCTION__.6278
-  cc:	00000098 	.word	0x00000098
-			cc: R_ARM_REL32	.LC7
-  d0:	0000008c 	.word	0x0000008c
-			d0: R_ARM_REL32	.LC6
-  d4:	0000003c 	.word	0x0000003c
-			d4: R_ARM_REL32	.rodata.__FUNCTION__.6278
-  d8:	0000003c 	.word	0x0000003c
-			d8: R_ARM_REL32	.LC6
-  dc:	00000020 	.word	0x00000020
-			dc: R_ARM_REL32	.LC8
+  c8:	00000094 	.word	0x00000094
+			c8: R_ARM_REL32	.rodata.__FUNCTION__.6330
+  cc:	00000094 	.word	0x00000094
+			cc: R_ARM_REL32	.LC6
+  d0:	00000048 	.word	0x00000048
+			d0: R_ARM_REL32	.rodata.__FUNCTION__.6330
+  d4:	00000040 	.word	0x00000040
+			d4: R_ARM_REL32	.LC7
+  d8:	00000020 	.word	0x00000020
+			d8: R_ARM_REL32	.LC8
 
-Disassembly of section .text.DL_HI_Audio_Dlclose:
+Disassembly of section .text.DL_hi_audio_dlclose:
 
-00000000 <DL_HI_Audio_Dlclose>:
-   0:	e59f3054 	ldr	r3, [pc, #84]	; 5c <DL_HI_Audio_Dlclose+0x5c>
+00000000 <DL_hi_audio_dlclose>:
+   0:	e59f3044 	ldr	r3, [pc, #68]	; 4c <DL_hi_audio_dlclose+0x4c>
    4:	e3500000 	cmp	r0, #0
-   8:	e92d4007 	push	{r0, r1, r2, lr}
+   8:	e92d4010 	push	{r4, lr}
    c:	e08f3003 	add	r3, pc, r3
-  10:	1a00000e 	bne	50 <DL_HI_Audio_Dlclose+0x50>
-  14:	e59f2044 	ldr	r2, [pc, #68]	; 60 <DL_HI_Audio_Dlclose+0x60>
-  18:	e59f1044 	ldr	r1, [pc, #68]	; 64 <DL_HI_Audio_Dlclose+0x64>
+  10:	1a00000a 	bne	40 <DL_hi_audio_dlclose+0x40>
+  14:	e59f2034 	ldr	r2, [pc, #52]	; 50 <DL_hi_audio_dlclose+0x50>
+  18:	e59f1034 	ldr	r1, [pc, #52]	; 54 <DL_hi_audio_dlclose+0x54>
   1c:	e7930002 	ldr	r0, [r3, r2]
-  20:	e59f3040 	ldr	r3, [pc, #64]	; 68 <DL_HI_Audio_Dlclose+0x68>
-  24:	e59f2040 	ldr	r2, [pc, #64]	; 6c <DL_HI_Audio_Dlclose+0x6c>
-  28:	e08f3003 	add	r3, pc, r3
-  2c:	e58d3000 	str	r3, [sp]
-  30:	e3a03059 	mov	r3, #89	; 0x59
-  34:	e08f2002 	add	r2, pc, r2
-  38:	e08f1001 	add	r1, pc, r1
-  3c:	e5900000 	ldr	r0, [r0]
-  40:	ebfffffe 	bl	0 <fprintf>
-			40: R_ARM_CALL	fprintf
-  44:	e3e00000 	mvn	r0, #0
-  48:	e28dd00c 	add	sp, sp, #12
-  4c:	e49df004 	pop	{pc}		; (ldr pc, [sp], #4)
-  50:	ebfffffe 	bl	0 <dlclose>
-			50: R_ARM_CALL	dlclose
-  54:	e3a00000 	mov	r0, #0
-  58:	eafffffa 	b	48 <DL_HI_Audio_Dlclose+0x48>
-  5c:	00000048 	.word	0x00000048
-			5c: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
-  60:	00000000 	.word	0x00000000
-			60: R_ARM_GOT32	stderr
-  64:	00000024 	.word	0x00000024
-			64: R_ARM_REL32	.LC6
-  68:	00000038 	.word	0x00000038
-			68: R_ARM_REL32	.LC7
-  6c:	00000030 	.word	0x00000030
-			6c: R_ARM_REL32	.rodata.__FUNCTION__.6282
+  20:	e3a03058 	mov	r3, #88	; 0x58
+  24:	e59f202c 	ldr	r2, [pc, #44]	; 58 <DL_hi_audio_dlclose+0x58>
+  28:	e08f1001 	add	r1, pc, r1
+  2c:	e08f2002 	add	r2, pc, r2
+  30:	e5900000 	ldr	r0, [r0]
+  34:	ebfffffe 	bl	0 <fprintf>
+			34: R_ARM_CALL	fprintf
+  38:	e3e00000 	mvn	r0, #0
+  3c:	e8bd8010 	pop	{r4, pc}
+  40:	ebfffffe 	bl	0 <dlclose>
+			40: R_ARM_CALL	dlclose
+  44:	e3a00000 	mov	r0, #0
+  48:	e8bd8010 	pop	{r4, pc}
+  4c:	00000038 	.word	0x00000038
+			4c: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
+  50:	00000000 	.word	0x00000000
+			50: R_ARM_GOT32	stderr
+  54:	00000024 	.word	0x00000024
+			54: R_ARM_REL32	.LC6
+  58:	00000024 	.word	0x00000024
+			58: R_ARM_REL32	.rodata.__FUNCTION__.6334
