--- sdk/vendor-disasm/v2_0_2_0/libive/ive_queue.S	2026-03-29 07:34:58.951973148 +0300
+++ sdk/vendor-disasm/v2_0_2_1/libive/ive_queue.S	2026-03-29 07:34:59.841975576 +0300
@@ -7,11 +7,12 @@
 00000000 l    d  .data	00000000 .data
 00000000 l    d  .bss	00000000 .bss
 00000000 l    d  .text.ive_create_obj_list	00000000 .text.ive_create_obj_list
-00000000 l       .rodata.ive_create_obj_list.str1.4	00000000 .LC0
 0000003c l       .rodata.ive_create_obj_list.str1.4	00000000 .LC1
+0000006c l       .rodata.ive_create_obj_list.str1.4	00000000 .LC2
+00000000 l       .rodata.ive_create_obj_list.str1.4	00000000 .LC0
 00000000 l    d  .text.ive_destroy_obj_list	00000000 .text.ive_destroy_obj_list
 00000000 l    d  .text.ive_obj_list_get_free	00000000 .text.ive_obj_list_get_free
-00000000 l       .rodata.ive_obj_list_get_free.str1.4	00000000 .LC2
+00000000 l       .rodata.ive_obj_list_get_free.str1.4	00000000 .LC3
 00000000 l    d  .text.ive_obj_list_put_free	00000000 .text.ive_obj_list_put_free
 00000000 l    d  .text.ive_obj_list_get_free_num	00000000 .text.ive_obj_list_get_free_num
 00000000 l    d  .text.ive_obj_list_get_train_num	00000000 .text.ive_obj_list_get_train_num
@@ -19,29 +20,29 @@
 00000000 l    d  .text.ive_obj_list_put_track	00000000 .text.ive_obj_list_put_track
 00000000 l    d  .text.ive_obj_list_put_train	00000000 .text.ive_obj_list_put_train
 00000000 l    d  .text.ive_obj_list_get_train	00000000 .text.ive_obj_list_get_train
-00000000 l       .rodata.ive_obj_list_get_train.str1.4	00000000 .LC3
+00000000 l       .rodata.ive_obj_list_get_train.str1.4	00000000 .LC4
 00000000 l    d  .text.ive_obj_list_query_train	00000000 .text.ive_obj_list_query_train
-00000000 l       .rodata.ive_obj_list_query_train.str1.4	00000000 .LC4
+00000000 l       .rodata.ive_obj_list_query_train.str1.4	00000000 .LC5
 00000000 l    d  .text.ive_obj_list_get_track	00000000 .text.ive_obj_list_get_track
-00000000 l       .rodata.ive_obj_list_get_track.str1.4	00000000 .LC5
+00000000 l       .rodata.ive_obj_list_get_track.str1.4	00000000 .LC6
 00000000 l    d  .text.ive_obj_list_query_track	00000000 .text.ive_obj_list_query_track
-00000000 l       .rodata.ive_obj_list_query_track.str1.4	00000000 .LC6
+00000000 l       .rodata.ive_obj_list_query_track.str1.4	00000000 .LC7
 00000000 l    d  .text.ive_obj_free_track_node_by_node	00000000 .text.ive_obj_free_track_node_by_node
-00000000 l       .rodata.ive_obj_free_track_node_by_node.str1.4	00000000 .LC7
-00000000 l    d  .rodata.__FUNCTION__.9370	00000000 .rodata.__FUNCTION__.9370
-00000000 l     O .rodata.__FUNCTION__.9370	00000014 __FUNCTION__.9370
-00000000 l    d  .rodata.__FUNCTION__.9379	00000000 .rodata.__FUNCTION__.9379
-00000000 l     O .rodata.__FUNCTION__.9379	00000016 __FUNCTION__.9379
-00000000 l    d  .rodata.__FUNCTION__.9406	00000000 .rodata.__FUNCTION__.9406
-00000000 l     O .rodata.__FUNCTION__.9406	00000017 __FUNCTION__.9406
-00000000 l    d  .rodata.__FUNCTION__.9415	00000000 .rodata.__FUNCTION__.9415
-00000000 l     O .rodata.__FUNCTION__.9415	00000019 __FUNCTION__.9415
-00000000 l    d  .rodata.__FUNCTION__.9424	00000000 .rodata.__FUNCTION__.9424
-00000000 l     O .rodata.__FUNCTION__.9424	00000017 __FUNCTION__.9424
-00000000 l    d  .rodata.__FUNCTION__.9433	00000000 .rodata.__FUNCTION__.9433
-00000000 l     O .rodata.__FUNCTION__.9433	00000019 __FUNCTION__.9433
-00000000 l    d  .rodata.__FUNCTION__.9445	00000000 .rodata.__FUNCTION__.9445
-00000000 l     O .rodata.__FUNCTION__.9445	00000020 __FUNCTION__.9445
+00000000 l       .rodata.ive_obj_free_track_node_by_node.str1.4	00000000 .LC8
+00000000 l    d  .rodata.__FUNCTION__.9327	00000000 .rodata.__FUNCTION__.9327
+00000000 l     O .rodata.__FUNCTION__.9327	00000014 __FUNCTION__.9327
+00000000 l    d  .rodata.__FUNCTION__.9338	00000000 .rodata.__FUNCTION__.9338
+00000000 l     O .rodata.__FUNCTION__.9338	00000016 __FUNCTION__.9338
+00000000 l    d  .rodata.__FUNCTION__.9365	00000000 .rodata.__FUNCTION__.9365
+00000000 l     O .rodata.__FUNCTION__.9365	00000017 __FUNCTION__.9365
+00000000 l    d  .rodata.__FUNCTION__.9374	00000000 .rodata.__FUNCTION__.9374
+00000000 l     O .rodata.__FUNCTION__.9374	00000019 __FUNCTION__.9374
+00000000 l    d  .rodata.__FUNCTION__.9383	00000000 .rodata.__FUNCTION__.9383
+00000000 l     O .rodata.__FUNCTION__.9383	00000017 __FUNCTION__.9383
+00000000 l    d  .rodata.__FUNCTION__.9392	00000000 .rodata.__FUNCTION__.9392
+00000000 l     O .rodata.__FUNCTION__.9392	00000019 __FUNCTION__.9392
+00000000 l    d  .rodata.__FUNCTION__.9404	00000000 .rodata.__FUNCTION__.9404
+00000000 l     O .rodata.__FUNCTION__.9404	00000020 __FUNCTION__.9404
 00000000 l    d  .rodata.ive_create_obj_list.str1.4	00000000 .rodata.ive_create_obj_list.str1.4
 00000000 l    d  .rodata.ive_obj_free_track_node_by_node.str1.4	00000000 .rodata.ive_obj_free_track_node_by_node.str1.4
 00000000 l    d  .rodata.ive_obj_list_get_free.str1.4	00000000 .rodata.ive_obj_list_get_free.str1.4
@@ -52,14 +53,14 @@
 00000000 l    d  .note.GNU-stack	00000000 .note.GNU-stack
 00000000 l    d  .comment	00000000 .comment
 00000000 l    d  .ARM.attributes	00000000 .ARM.attributes
-00000000 g     F .text.ive_create_obj_list	000002b0 ive_create_obj_list
+00000000 g     F .text.ive_create_obj_list	00000368 ive_create_obj_list
 00000000         *UND*	00000000 malloc
 00000000         *UND*	00000000 memset_s
 00000000         *UND*	00000000 fprintf
 00000000         *UND*	00000000 free
 00000000         *UND*	00000000 _GLOBAL_OFFSET_TABLE_
 00000000         *UND*	00000000 stderr
-00000000 g     F .text.ive_destroy_obj_list	00000070 ive_destroy_obj_list
+00000000 g     F .text.ive_destroy_obj_list	00000078 ive_destroy_obj_list
 00000000 g     F .text.ive_obj_list_get_free	00000098 ive_obj_list_get_free
 00000000 g     F .text.ive_obj_list_put_free	00000028 ive_obj_list_put_free
 00000000 g     F .text.ive_obj_list_get_free_num	00000008 ive_obj_list_get_free_num
@@ -79,190 +80,243 @@
 
 00000000 <ive_create_obj_list>:
    0:	e92d4ff0 	push	{r4, r5, r6, r7, r8, r9, sl, fp, lr}
-   4:	e3a070b8 	mov	r7, #184	; 0xb8
-   8:	e59f9288 	ldr	r9, [pc, #648]	; 298 <ive_create_obj_list+0x298>
-   c:	e24dd03c 	sub	sp, sp, #60	; 0x3c
-  10:	e0070297 	mul	r7, r7, r2
-  14:	e58d202c 	str	r2, [sp, #44]	; 0x2c
-  18:	e2815004 	add	r5, r1, #4
-  1c:	e5815004 	str	r5, [r1, #4]
-  20:	e281200c 	add	r2, r1, #12
-  24:	e5815008 	str	r5, [r1, #8]
-  28:	e2813014 	add	r3, r1, #20
+   4:	e3a060b8 	mov	r6, #184	; 0xb8
+   8:	e0060296 	mul	r6, r6, r2
+   c:	e59f932c 	ldr	r9, [pc, #812]	; 340 <ive_create_obj_list+0x340>
+  10:	e2815004 	add	r5, r1, #4
+  14:	e5815004 	str	r5, [r1, #4]
+  18:	e2813014 	add	r3, r1, #20
+  1c:	e5815008 	str	r5, [r1, #8]
+  20:	e1a07002 	mov	r7, r2
+  24:	e5813014 	str	r3, [r1, #20]
+  28:	e281200c 	add	r2, r1, #12
   2c:	e581200c 	str	r2, [r1, #12]
-  30:	e1a06000 	mov	r6, r0
+  30:	e24dd03c 	sub	sp, sp, #60	; 0x3c
   34:	e5812010 	str	r2, [r1, #16]
-  38:	e1a00007 	mov	r0, r7
-  3c:	e5813014 	str	r3, [r1, #20]
-  40:	e1a04001 	mov	r4, r1
-  44:	e5813018 	str	r3, [r1, #24]
-  48:	e08f9009 	add	r9, pc, r9
+  38:	e1a08000 	mov	r8, r0
+  3c:	e5813018 	str	r3, [r1, #24]
+  40:	e1a00006 	mov	r0, r6
+  44:	e08f9009 	add	r9, pc, r9
+  48:	e1a04001 	mov	r4, r1
   4c:	ebfffffe 	bl	0 <malloc>
 			4c: R_ARM_CALL	malloc
   50:	e5840000 	str	r0, [r4]
   54:	e3500000 	cmp	r0, #0
-  58:	0a000073 	beq	22c <ive_create_obj_list+0x22c>
-  5c:	e1a03007 	mov	r3, r7
-  60:	e1a01007 	mov	r1, r7
+  58:	0a0000ac 	beq	310 <ive_create_obj_list+0x310>
+  5c:	e1a03006 	mov	r3, r6
+  60:	e1a01006 	mov	r1, r6
   64:	e3a02000 	mov	r2, #0
   68:	ebfffffe 	bl	0 <memset_s>
 			68: R_ARM_CALL	memset_s
-  6c:	e3050880 	movw	r0, #22656	; 0x5880
-  70:	ebfffffe 	bl	0 <malloc>
-			70: R_ARM_CALL	malloc
-  74:	e5840030 	str	r0, [r4, #48]	; 0x30
-  78:	e3500000 	cmp	r0, #0
-  7c:	e1a07000 	mov	r7, r0
-  80:	0a000075 	beq	25c <ive_create_obj_list+0x25c>
-  84:	e3053880 	movw	r3, #22656	; 0x5880
-  88:	e3a02000 	mov	r2, #0
-  8c:	e1a01003 	mov	r1, r3
-  90:	ebfffffe 	bl	0 <memset_s>
-			90: R_ARM_CALL	memset_s
-  94:	e59dc02c 	ldr	ip, [sp, #44]	; 0x2c
-  98:	e3a0ecba 	mov	lr, #47616	; 0xba00
-  9c:	e1c680d0 	ldrd	r8, [r6]
-  a0:	e0000c9e 	mul	r0, lr, ip
-  a4:	e1c660d8 	ldrd	r6, [r6, #8]
-  a8:	e098a000 	adds	sl, r8, r0
-  ac:	e3a01000 	mov	r1, #0
-  b0:	e0a9b001 	adc	fp, r9, r1
-  b4:	e1cd60f8 	strd	r6, [sp, #8]
-  b8:	e0966000 	adds	r6, r6, r0
-  bc:	e0a77001 	adc	r7, r7, r1
-  c0:	e1a00006 	mov	r0, r6
-  c4:	e1a0268c 	lsl	r2, ip, #13
-  c8:	e1a01007 	mov	r1, r7
-  cc:	e09a6002 	adds	r6, sl, r2
-  d0:	e1cd80f0 	strd	r8, [sp]
-  d4:	e5943000 	ldr	r3, [r4]
-  d8:	e3a09000 	mov	r9, #0
-  dc:	e1cd01f0 	strd	r0, [sp, #16]
-  e0:	e0ab7009 	adc	r7, fp, r9
-  e4:	e0900002 	adds	r0, r0, r2
-  e8:	e0a11009 	adc	r1, r1, r9
-  ec:	e35c0000 	cmp	ip, #0
-  f0:	0a000044 	beq	208 <ive_create_obj_list+0x208>
-  f4:	e0500006 	subs	r0, r0, r6
-  f8:	e0c11007 	sbc	r1, r1, r7
-  fc:	e1a0c00e 	mov	ip, lr
- 100:	e59d202c 	ldr	r2, [sp, #44]	; 0x2c
- 104:	e1cd02f0 	strd	r0, [sp, #32]
- 108:	e1cd01d0 	ldrd	r0, [sp, #16]
- 10c:	e2428001 	sub	r8, r2, #1
- 110:	e3a02a02 	mov	r2, #8192	; 0x2000
- 114:	e050000a 	subs	r0, r0, sl
- 118:	e0c1100b 	sbc	r1, r1, fp
- 11c:	e58d8028 	str	r8, [sp, #40]	; 0x28
- 120:	e1cd01f8 	strd	r0, [sp, #24]
- 124:	e1cd80d0 	ldrd	r8, [sp]
- 128:	e1cd00d8 	ldrd	r0, [sp, #8]
- 12c:	e1cd83f0 	strd	r8, [sp, #48]	; 0x30
- 130:	e0508008 	subs	r8, r0, r8
- 134:	e0c19009 	sbc	r9, r1, r9
- 138:	e1a00008 	mov	r0, r8
- 13c:	e1a01009 	mov	r1, r9
- 140:	e1cd01f0 	strd	r0, [sp, #16]
- 144:	e1cd00d0 	ldrd	r0, [sp]
- 148:	e59d8028 	ldr	r8, [sp, #40]	; 0x28
- 14c:	e58d4000 	str	r4, [sp]
- 150:	e2900cba 	adds	r0, r0, #47616	; 0xba00
- 154:	e2a11000 	adc	r1, r1, #0
- 158:	e1a04006 	mov	r4, r6
- 15c:	e0a1089e 	umlal	r0, r1, lr, r8
- 160:	e58d5028 	str	r5, [sp, #40]	; 0x28
- 164:	e3a0e010 	mov	lr, #16
- 168:	e1cd83d0 	ldrd	r8, [sp, #48]	; 0x30
- 16c:	e1a05007 	mov	r5, r7
- 170:	e1cd00f8 	strd	r0, [sp, #8]
- 174:	e1cd01d0 	ldrd	r0, [sp, #16]
- 178:	e1cd61d8 	ldrd	r6, [sp, #24]
- 17c:	e0900008 	adds	r0, r0, r8
- 180:	e1c386f8 	strd	r8, [r3, #104]	; 0x68
- 184:	e0a11009 	adc	r1, r1, r9
- 188:	e2988cba 	adds	r8, r8, #47616	; 0xba00
- 18c:	e1c307f0 	strd	r0, [r3, #112]	; 0x70
- 190:	e2a99000 	adc	r9, r9, #0
- 194:	e1cd02d0 	ldrd	r0, [sp, #32]
- 198:	e096600a 	adds	r6, r6, sl
- 19c:	e1c3a8f0 	strd	sl, [r3, #128]	; 0x80
- 1a0:	e0a7700b 	adc	r7, r7, fp
- 1a4:	e29aaa02 	adds	sl, sl, #8192	; 0x2000
- 1a8:	e1c368f8 	strd	r6, [r3, #136]	; 0x88
- 1ac:	e2abb000 	adc	fp, fp, #0
- 1b0:	e1cd60d8 	ldrd	r6, [sp, #8]
- 1b4:	e0900004 	adds	r0, r0, r4
- 1b8:	e1c349f8 	strd	r4, [r3, #152]	; 0x98
- 1bc:	e0a11005 	adc	r1, r1, r5
- 1c0:	e2944010 	adds	r4, r4, #16
- 1c4:	e2a55000 	adc	r5, r5, #0
- 1c8:	e1590007 	cmp	r9, r7
- 1cc:	e59d7000 	ldr	r7, [sp]
- 1d0:	01580006 	cmpeq	r8, r6
- 1d4:	e5976008 	ldr	r6, [r7, #8]
- 1d8:	e1c30af0 	strd	r0, [r3, #160]	; 0xa0
- 1dc:	e59d1028 	ldr	r1, [sp, #40]	; 0x28
- 1e0:	e583c078 	str	ip, [r3, #120]	; 0x78
- 1e4:	e5832090 	str	r2, [r3, #144]	; 0x90
- 1e8:	e583e0a8 	str	lr, [r3, #168]	; 0xa8
- 1ec:	e5873008 	str	r3, [r7, #8]
- 1f0:	e5836004 	str	r6, [r3, #4]
- 1f4:	e5831000 	str	r1, [r3]
- 1f8:	e5863000 	str	r3, [r6]
- 1fc:	e28330b8 	add	r3, r3, #184	; 0xb8
- 200:	1affffdb 	bne	174 <ive_create_obj_list+0x174>
- 204:	e1a04007 	mov	r4, r7
- 208:	e59d202c 	ldr	r2, [sp, #44]	; 0x2c
- 20c:	e3a03000 	mov	r3, #0
- 210:	e5843020 	str	r3, [r4, #32]
- 214:	e1a00003 	mov	r0, r3
- 218:	e584201c 	str	r2, [r4, #28]
- 21c:	e5842028 	str	r2, [r4, #40]	; 0x28
- 220:	e5843024 	str	r3, [r4, #36]	; 0x24
- 224:	e28dd03c 	add	sp, sp, #60	; 0x3c
- 228:	e8bd8ff0 	pop	{r4, r5, r6, r7, r8, r9, sl, fp, pc}
- 22c:	e59f1068 	ldr	r1, [pc, #104]	; 29c <ive_create_obj_list+0x29c>
- 230:	e3a0304f 	mov	r3, #79	; 0x4f
- 234:	e59f2064 	ldr	r2, [pc, #100]	; 2a0 <ive_create_obj_list+0x2a0>
- 238:	e7990001 	ldr	r0, [r9, r1]
- 23c:	e59f1060 	ldr	r1, [pc, #96]	; 2a4 <ive_create_obj_list+0x2a4>
- 240:	e08f2002 	add	r2, pc, r2
- 244:	e08f1001 	add	r1, pc, r1
- 248:	e5900000 	ldr	r0, [r0]
- 24c:	ebfffffe 	bl	0 <fprintf>
-			24c: R_ARM_CALL	fprintf
- 250:	e308000d 	movw	r0, #32781	; 0x800d
- 254:	e34a001d 	movt	r0, #40989	; 0xa01d
- 258:	eafffff1 	b	224 <ive_create_obj_list+0x224>
- 25c:	e59f1038 	ldr	r1, [pc, #56]	; 29c <ive_create_obj_list+0x29c>
- 260:	e3a03057 	mov	r3, #87	; 0x57
- 264:	e59f203c 	ldr	r2, [pc, #60]	; 2a8 <ive_create_obj_list+0x2a8>
- 268:	e7990001 	ldr	r0, [r9, r1]
- 26c:	e59f1038 	ldr	r1, [pc, #56]	; 2ac <ive_create_obj_list+0x2ac>
- 270:	e08f2002 	add	r2, pc, r2
- 274:	e08f1001 	add	r1, pc, r1
- 278:	e5900000 	ldr	r0, [r0]
- 27c:	ebfffffe 	bl	0 <fprintf>
-			27c: R_ARM_CALL	fprintf
- 280:	e5940000 	ldr	r0, [r4]
- 284:	ebfffffe 	bl	0 <free>
-			284: R_ARM_CALL	free
- 288:	e5847000 	str	r7, [r4]
- 28c:	e308000d 	movw	r0, #32781	; 0x800d
- 290:	e34a001d 	movt	r0, #40989	; 0xa01d
- 294:	eaffffe2 	b	224 <ive_create_obj_list+0x224>
- 298:	00000248 	.word	0x00000248
-			298: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
- 29c:	00000000 	.word	0x00000000
-			29c: R_ARM_GOT32	stderr
- 2a0:	00000058 	.word	0x00000058
-			2a0: R_ARM_REL32	.rodata.__FUNCTION__.9370
- 2a4:	00000058 	.word	0x00000058
-			2a4: R_ARM_REL32	.LC0
- 2a8:	00000030 	.word	0x00000030
-			2a8: R_ARM_REL32	.rodata.__FUNCTION__.9370
- 2ac:	00000030 	.word	0x00000030
-			2ac: R_ARM_REL32	.LC1
+  6c:	e250a000 	subs	sl, r0, #0
+  70:	1a00001f 	bne	f4 <ive_create_obj_list+0xf4>
+  74:	e3050880 	movw	r0, #22656	; 0x5880
+  78:	ebfffffe 	bl	0 <malloc>
+			78: R_ARM_CALL	malloc
+  7c:	e5840030 	str	r0, [r4, #48]	; 0x30
+  80:	e3500000 	cmp	r0, #0
+  84:	0a000095 	beq	2e0 <ive_create_obj_list+0x2e0>
+  88:	e3053880 	movw	r3, #22656	; 0x5880
+  8c:	e1a0200a 	mov	r2, sl
+  90:	e1a01003 	mov	r1, r3
+  94:	ebfffffe 	bl	0 <memset_s>
+			94: R_ARM_CALL	memset_s
+  98:	e2506000 	subs	r6, r0, #0
+  9c:	0a000020 	beq	124 <ive_create_obj_list+0x124>
+  a0:	e59f129c 	ldr	r1, [pc, #668]	; 344 <ive_create_obj_list+0x344>
+  a4:	e3086003 	movw	r6, #32771	; 0x8003
+  a8:	e59f2298 	ldr	r2, [pc, #664]	; 348 <ive_create_obj_list+0x348>
+  ac:	e3a0305f 	mov	r3, #95	; 0x5f
+  b0:	e7990001 	ldr	r0, [r9, r1]
+  b4:	e59f1290 	ldr	r1, [pc, #656]	; 34c <ive_create_obj_list+0x34c>
+  b8:	e08f2002 	add	r2, pc, r2
+  bc:	e08f1001 	add	r1, pc, r1
+  c0:	e5900000 	ldr	r0, [r0]
+  c4:	e34a601d 	movt	r6, #40989	; 0xa01d
+  c8:	ebfffffe 	bl	0 <fprintf>
+			c8: R_ARM_CALL	fprintf
+  cc:	e5940030 	ldr	r0, [r4, #48]	; 0x30
+  d0:	ebfffffe 	bl	0 <free>
+			d0: R_ARM_CALL	free
+  d4:	e584a030 	str	sl, [r4, #48]	; 0x30
+  d8:	e5940000 	ldr	r0, [r4]
+  dc:	ebfffffe 	bl	0 <free>
+			dc: R_ARM_CALL	free
+  e0:	e3a03000 	mov	r3, #0
+  e4:	e5843000 	str	r3, [r4]
+  e8:	e1a00006 	mov	r0, r6
+  ec:	e28dd03c 	add	sp, sp, #60	; 0x3c
+  f0:	e8bd8ff0 	pop	{r4, r5, r6, r7, r8, r9, sl, fp, pc}
+  f4:	e59f1248 	ldr	r1, [pc, #584]	; 344 <ive_create_obj_list+0x344>
+  f8:	e3a03052 	mov	r3, #82	; 0x52
+  fc:	e59f224c 	ldr	r2, [pc, #588]	; 350 <ive_create_obj_list+0x350>
+ 100:	e3086003 	movw	r6, #32771	; 0x8003
+ 104:	e7990001 	ldr	r0, [r9, r1]
+ 108:	e59f1244 	ldr	r1, [pc, #580]	; 354 <ive_create_obj_list+0x354>
+ 10c:	e08f2002 	add	r2, pc, r2
+ 110:	e08f1001 	add	r1, pc, r1
+ 114:	e5900000 	ldr	r0, [r0]
+ 118:	e34a601d 	movt	r6, #40989	; 0xa01d
+ 11c:	ebfffffe 	bl	0 <fprintf>
+			11c: R_ARM_CALL	fprintf
+ 120:	eaffffec 	b	d8 <ive_create_obj_list+0xd8>
+ 124:	e1c8a0d0 	ldrd	sl, [r8]
+ 128:	e594c000 	ldr	ip, [r4]
+ 12c:	e3a03cba 	mov	r3, #47616	; 0xba00
+ 130:	e0020793 	mul	r2, r3, r7
+ 134:	e1c880d8 	ldrd	r8, [r8, #8]
+ 138:	e1a0000a 	mov	r0, sl
+ 13c:	e1a0100b 	mov	r1, fp
+ 140:	e3a03000 	mov	r3, #0
+ 144:	e1a0a008 	mov	sl, r8
+ 148:	e0908002 	adds	r8, r0, r2
+ 14c:	e1a0b009 	mov	fp, r9
+ 150:	e1cd00f0 	strd	r0, [sp]
+ 154:	e0a19003 	adc	r9, r1, r3
+ 158:	e1cda0f8 	strd	sl, [sp, #8]
+ 15c:	e1a00687 	lsl	r0, r7, #13
+ 160:	e09aa002 	adds	sl, sl, r2
+ 164:	e1a02000 	mov	r2, r0
+ 168:	e0abb003 	adc	fp, fp, r3
+ 16c:	e3a03000 	mov	r3, #0
+ 170:	e0980000 	adds	r0, r8, r0
+ 174:	e0a91003 	adc	r1, r9, r3
+ 178:	e1cda1f0 	strd	sl, [sp, #16]
+ 17c:	e09aa002 	adds	sl, sl, r2
+ 180:	e0abb003 	adc	fp, fp, r3
+ 184:	e3570000 	cmp	r7, #0
+ 188:	ec410b30 	vmov	d16, r0, r1
+ 18c:	e1a0200a 	mov	r2, sl
+ 190:	e1a0300b 	mov	r3, fp
+ 194:	0a000047 	beq	2b8 <ive_create_obj_list+0x2b8>
+ 198:	ec5bab30 	vmov	sl, fp, d16
+ 19c:	e2471001 	sub	r1, r7, #1
+ 1a0:	e58d1030 	str	r1, [sp, #48]	; 0x30
+ 1a4:	e3a0ea02 	mov	lr, #8192	; 0x2000
+ 1a8:	e052a00a 	subs	sl, r2, sl
+ 1ac:	e0c3b00b 	sbc	fp, r3, fp
+ 1b0:	e1cd21d0 	ldrd	r2, [sp, #16]
+ 1b4:	e1cd00d0 	ldrd	r0, [sp]
+ 1b8:	e0522008 	subs	r2, r2, r8
+ 1bc:	e0c33009 	sbc	r3, r3, r9
+ 1c0:	e1cda2f0 	strd	sl, [sp, #32]
+ 1c4:	e1cd21f8 	strd	r2, [sp, #24]
+ 1c8:	e58d7034 	str	r7, [sp, #52]	; 0x34
+ 1cc:	e1a0a000 	mov	sl, r0
+ 1d0:	e1cd20d8 	ldrd	r2, [sp, #8]
+ 1d4:	e1a0b001 	mov	fp, r1
+ 1d8:	edcd0b0a 	vstr	d16, [sp, #40]	; 0x28
+ 1dc:	e052a00a 	subs	sl, r2, sl
+ 1e0:	e0c3b00b 	sbc	fp, r3, fp
+ 1e4:	e1a0200a 	mov	r2, sl
+ 1e8:	e1a0300b 	mov	r3, fp
+ 1ec:	e59da030 	ldr	sl, [sp, #48]	; 0x30
+ 1f0:	e3a0bcba 	mov	fp, #47616	; 0xba00
+ 1f4:	e1cd21f0 	strd	r2, [sp, #16]
+ 1f8:	e58d6030 	str	r6, [sp, #48]	; 0x30
+ 1fc:	e2902cba 	adds	r2, r0, #47616	; 0xba00
+ 200:	e2a13000 	adc	r3, r1, #0
+ 204:	e0a32a9b 	umlal	r2, r3, fp, sl
+ 208:	ec576b30 	vmov	r6, r7, d16
+ 20c:	e1a0a000 	mov	sl, r0
+ 210:	e1a0b001 	mov	fp, r1
+ 214:	e1cd20f8 	strd	r2, [sp, #8]
+ 218:	e1cd21d0 	ldrd	r2, [sp, #16]
+ 21c:	e1cd01d8 	ldrd	r0, [sp, #24]
+ 220:	e092200a 	adds	r2, r2, sl
+ 224:	e1cca6f8 	strd	sl, [ip, #104]	; 0x68
+ 228:	e0a3300b 	adc	r3, r3, fp
+ 22c:	e29aacba 	adds	sl, sl, #47616	; 0xba00
+ 230:	e2abb000 	adc	fp, fp, #0
+ 234:	e0900008 	adds	r0, r0, r8
+ 238:	e0a11009 	adc	r1, r1, r9
+ 23c:	e1cc27f0 	strd	r2, [ip, #112]	; 0x70
+ 240:	e1cd00f0 	strd	r0, [sp]
+ 244:	e1cd22d0 	ldrd	r2, [sp, #32]
+ 248:	e58ce090 	str	lr, [ip, #144]	; 0x90
+ 24c:	e3a01cba 	mov	r1, #47616	; 0xba00
+ 250:	e1cc88f0 	strd	r8, [ip, #128]	; 0x80
+ 254:	e58c1078 	str	r1, [ip, #120]	; 0x78
+ 258:	e2988a02 	adds	r8, r8, #8192	; 0x2000
+ 25c:	e2a99000 	adc	r9, r9, #0
+ 260:	e3a01010 	mov	r1, #16
+ 264:	e58c10a8 	str	r1, [ip, #168]	; 0xa8
+ 268:	e0922006 	adds	r2, r2, r6
+ 26c:	e1cd00d8 	ldrd	r0, [sp, #8]
+ 270:	e1cc69f8 	strd	r6, [ip, #152]	; 0x98
+ 274:	e0a33007 	adc	r3, r3, r7
+ 278:	e2966010 	adds	r6, r6, #16
+ 27c:	e2a77000 	adc	r7, r7, #0
+ 280:	e15b0001 	cmp	fp, r1
+ 284:	015a0000 	cmpeq	sl, r0
+ 288:	e1cd00d0 	ldrd	r0, [sp]
+ 28c:	e1cc08f8 	strd	r0, [ip, #136]	; 0x88
+ 290:	e5941008 	ldr	r1, [r4, #8]
+ 294:	e1cc2af0 	strd	r2, [ip, #160]	; 0xa0
+ 298:	e584c008 	str	ip, [r4, #8]
+ 29c:	e58c1004 	str	r1, [ip, #4]
+ 2a0:	e58c5000 	str	r5, [ip]
+ 2a4:	e581c000 	str	ip, [r1]
+ 2a8:	e28cc0b8 	add	ip, ip, #184	; 0xb8
+ 2ac:	1affffd9 	bne	218 <ive_create_obj_list+0x218>
+ 2b0:	e59d6030 	ldr	r6, [sp, #48]	; 0x30
+ 2b4:	e59d7034 	ldr	r7, [sp, #52]	; 0x34
+ 2b8:	e584701c 	str	r7, [r4, #28]
+ 2bc:	e3a03000 	mov	r3, #0
+ 2c0:	e5847028 	str	r7, [r4, #40]	; 0x28
+ 2c4:	e1a00006 	mov	r0, r6
+ 2c8:	e5843020 	str	r3, [r4, #32]
+ 2cc:	e5843024 	str	r3, [r4, #36]	; 0x24
+ 2d0:	e5843034 	str	r3, [r4, #52]	; 0x34
+ 2d4:	e5843038 	str	r3, [r4, #56]	; 0x38
+ 2d8:	e28dd03c 	add	sp, sp, #60	; 0x3c
+ 2dc:	e8bd8ff0 	pop	{r4, r5, r6, r7, r8, r9, sl, fp, pc}
+ 2e0:	e59f105c 	ldr	r1, [pc, #92]	; 344 <ive_create_obj_list+0x344>
+ 2e4:	e3a03059 	mov	r3, #89	; 0x59
+ 2e8:	e59f2068 	ldr	r2, [pc, #104]	; 358 <ive_create_obj_list+0x358>
+ 2ec:	e308600d 	movw	r6, #32781	; 0x800d
+ 2f0:	e7990001 	ldr	r0, [r9, r1]
+ 2f4:	e59f1060 	ldr	r1, [pc, #96]	; 35c <ive_create_obj_list+0x35c>
+ 2f8:	e08f2002 	add	r2, pc, r2
+ 2fc:	e08f1001 	add	r1, pc, r1
+ 300:	e5900000 	ldr	r0, [r0]
+ 304:	e34a601d 	movt	r6, #40989	; 0xa01d
+ 308:	ebfffffe 	bl	0 <fprintf>
+			308: R_ARM_CALL	fprintf
+ 30c:	eaffff71 	b	d8 <ive_create_obj_list+0xd8>
+ 310:	e59f102c 	ldr	r1, [pc, #44]	; 344 <ive_create_obj_list+0x344>
+ 314:	e3a0304d 	mov	r3, #77	; 0x4d
+ 318:	e59f2040 	ldr	r2, [pc, #64]	; 360 <ive_create_obj_list+0x360>
+ 31c:	e308600d 	movw	r6, #32781	; 0x800d
+ 320:	e7990001 	ldr	r0, [r9, r1]
+ 324:	e59f1038 	ldr	r1, [pc, #56]	; 364 <ive_create_obj_list+0x364>
+ 328:	e08f2002 	add	r2, pc, r2
+ 32c:	e08f1001 	add	r1, pc, r1
+ 330:	e5900000 	ldr	r0, [r0]
+ 334:	e34a601d 	movt	r6, #40989	; 0xa01d
+ 338:	ebfffffe 	bl	0 <fprintf>
+			338: R_ARM_CALL	fprintf
+ 33c:	eaffff69 	b	e8 <ive_create_obj_list+0xe8>
+ 340:	000002f4 	.word	0x000002f4
+			340: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
+ 344:	00000000 	.word	0x00000000
+			344: R_ARM_GOT32	stderr
+ 348:	00000288 	.word	0x00000288
+			348: R_ARM_REL32	.rodata.__FUNCTION__.9327
+ 34c:	00000288 	.word	0x00000288
+			34c: R_ARM_REL32	.LC1
+ 350:	0000023c 	.word	0x0000023c
+			350: R_ARM_REL32	.rodata.__FUNCTION__.9327
+ 354:	0000023c 	.word	0x0000023c
+			354: R_ARM_REL32	.LC1
+ 358:	00000058 	.word	0x00000058
+			358: R_ARM_REL32	.rodata.__FUNCTION__.9327
+ 35c:	00000058 	.word	0x00000058
+			35c: R_ARM_REL32	.LC2
+ 360:	00000030 	.word	0x00000030
+			360: R_ARM_REL32	.rodata.__FUNCTION__.9327
+ 364:	00000030 	.word	0x00000030
+			364: R_ARM_REL32	.LC0
 
 Disassembly of section .text.ive_destroy_obj_list:
 
@@ -285,18 +339,20 @@
   3c:	e5845028 	str	r5, [r4, #40]	; 0x28
   40:	e5845020 	str	r5, [r4, #32]
   44:	e5845024 	str	r5, [r4, #36]	; 0x24
-  48:	0a000001 	beq	54 <ive_destroy_obj_list+0x54>
-  4c:	ebfffffe 	bl	0 <free>
-			4c: R_ARM_CALL	free
-  50:	e5845030 	str	r5, [r4, #48]	; 0x30
-  54:	e5940000 	ldr	r0, [r4]
-  58:	e3500000 	cmp	r0, #0
-  5c:	08bd8070 	popeq	{r4, r5, r6, pc}
-  60:	ebfffffe 	bl	0 <free>
-			60: R_ARM_CALL	free
-  64:	e3a03000 	mov	r3, #0
-  68:	e5843000 	str	r3, [r4]
-  6c:	e8bd8070 	pop	{r4, r5, r6, pc}
+  48:	e5845034 	str	r5, [r4, #52]	; 0x34
+  4c:	e5845038 	str	r5, [r4, #56]	; 0x38
+  50:	0a000001 	beq	5c <ive_destroy_obj_list+0x5c>
+  54:	ebfffffe 	bl	0 <free>
+			54: R_ARM_CALL	free
+  58:	e5845030 	str	r5, [r4, #48]	; 0x30
+  5c:	e5940000 	ldr	r0, [r4]
+  60:	e3500000 	cmp	r0, #0
+  64:	08bd8070 	popeq	{r4, r5, r6, pc}
+  68:	ebfffffe 	bl	0 <free>
+			68: R_ARM_CALL	free
+  6c:	e3a03000 	mov	r3, #0
+  70:	e5843000 	str	r3, [r4]
+  74:	e8bd8070 	pop	{r4, r5, r6, pc}
 
 Disassembly of section .text.ive_obj_list_get_free:
 
@@ -325,7 +381,7 @@
   54:	e581201c 	str	r2, [r1, #28]
   58:	e8bd8070 	pop	{r4, r5, r6, pc}
   5c:	e59f1028 	ldr	r1, [pc, #40]	; 8c <ive_obj_list_get_free+0x8c>
-  60:	e3a03086 	mov	r3, #134	; 0x86
+  60:	e3a03096 	mov	r3, #150	; 0x96
   64:	e59f2024 	ldr	r2, [pc, #36]	; 90 <ive_obj_list_get_free+0x90>
   68:	e79c0001 	ldr	r0, [ip, r1]
   6c:	e59f1020 	ldr	r1, [pc, #32]	; 94 <ive_obj_list_get_free+0x94>
@@ -341,9 +397,9 @@
   8c:	00000000 	.word	0x00000000
 			8c: R_ARM_GOT32	stderr
   90:	00000018 	.word	0x00000018
-			90: R_ARM_REL32	.rodata.__FUNCTION__.9379
+			90: R_ARM_REL32	.rodata.__FUNCTION__.9338
   94:	00000018 	.word	0x00000018
-			94: R_ARM_REL32	.LC2
+			94: R_ARM_REL32	.LC3
 
 Disassembly of section .text.ive_obj_list_put_free:
 
@@ -432,7 +488,7 @@
   54:	e5812020 	str	r2, [r1, #32]
   58:	e8bd8070 	pop	{r4, r5, r6, pc}
   5c:	e59f1028 	ldr	r1, [pc, #40]	; 8c <ive_obj_list_get_train+0x8c>
-  60:	e3a030c1 	mov	r3, #193	; 0xc1
+  60:	e3a030cd 	mov	r3, #205	; 0xcd
   64:	e59f2024 	ldr	r2, [pc, #36]	; 90 <ive_obj_list_get_train+0x90>
   68:	e79c0001 	ldr	r0, [ip, r1]
   6c:	e59f1020 	ldr	r1, [pc, #32]	; 94 <ive_obj_list_get_train+0x94>
@@ -448,9 +504,9 @@
   8c:	00000000 	.word	0x00000000
 			8c: R_ARM_GOT32	stderr
   90:	00000018 	.word	0x00000018
-			90: R_ARM_REL32	.rodata.__FUNCTION__.9406
+			90: R_ARM_REL32	.rodata.__FUNCTION__.9365
   94:	00000018 	.word	0x00000018
-			94: R_ARM_REL32	.LC3
+			94: R_ARM_REL32	.LC4
 
 Disassembly of section .text.ive_obj_list_query_train:
 
@@ -487,7 +543,7 @@
   74:	e59f2030 	ldr	r2, [pc, #48]	; ac <ive_obj_list_query_train+0xac>
   78:	e1a0c001 	mov	ip, r1
   7c:	e59f102c 	ldr	r1, [pc, #44]	; b0 <ive_obj_list_query_train+0xb0>
-  80:	e3a030d7 	mov	r3, #215	; 0xd7
+  80:	e3a030e3 	mov	r3, #227	; 0xe3
   84:	e08f2002 	add	r2, pc, r2
   88:	e79e0001 	ldr	r0, [lr, r1]
   8c:	e59f1020 	ldr	r1, [pc, #32]	; b4 <ive_obj_list_query_train+0xb4>
@@ -501,11 +557,11 @@
   a8:	0000007c 	.word	0x0000007c
 			a8: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
   ac:	00000020 	.word	0x00000020
-			ac: R_ARM_REL32	.rodata.__FUNCTION__.9415
+			ac: R_ARM_REL32	.rodata.__FUNCTION__.9374
   b0:	00000000 	.word	0x00000000
 			b0: R_ARM_GOT32	stderr
   b4:	00000018 	.word	0x00000018
-			b4: R_ARM_REL32	.LC4
+			b4: R_ARM_REL32	.LC5
 
 Disassembly of section .text.ive_obj_list_get_track:
 
@@ -534,7 +590,7 @@
   54:	e5812024 	str	r2, [r1, #36]	; 0x24
   58:	e8bd8070 	pop	{r4, r5, r6, pc}
   5c:	e59f1028 	ldr	r1, [pc, #40]	; 8c <ive_obj_list_get_track+0x8c>
-  60:	e3a030f2 	mov	r3, #242	; 0xf2
+  60:	e3a030fe 	mov	r3, #254	; 0xfe
   64:	e59f2024 	ldr	r2, [pc, #36]	; 90 <ive_obj_list_get_track+0x90>
   68:	e79c0001 	ldr	r0, [ip, r1]
   6c:	e59f1020 	ldr	r1, [pc, #32]	; 94 <ive_obj_list_get_track+0x94>
@@ -550,9 +606,9 @@
   8c:	00000000 	.word	0x00000000
 			8c: R_ARM_GOT32	stderr
   90:	00000018 	.word	0x00000018
-			90: R_ARM_REL32	.rodata.__FUNCTION__.9424
+			90: R_ARM_REL32	.rodata.__FUNCTION__.9383
   94:	00000018 	.word	0x00000018
-			94: R_ARM_REL32	.LC5
+			94: R_ARM_REL32	.LC6
 
 Disassembly of section .text.ive_obj_list_query_track:
 
@@ -589,7 +645,7 @@
   74:	e59f2030 	ldr	r2, [pc, #48]	; ac <ive_obj_list_query_track+0xac>
   78:	e1a0c001 	mov	ip, r1
   7c:	e59f102c 	ldr	r1, [pc, #44]	; b0 <ive_obj_list_query_track+0xb0>
-  80:	e3a03f42 	mov	r3, #264	; 0x108
+  80:	e3003112 	movw	r3, #274	; 0x112
   84:	e08f2002 	add	r2, pc, r2
   88:	e79e0001 	ldr	r0, [lr, r1]
   8c:	e59f1020 	ldr	r1, [pc, #32]	; b4 <ive_obj_list_query_track+0xb4>
@@ -603,11 +659,11 @@
   a8:	0000007c 	.word	0x0000007c
 			a8: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
   ac:	00000020 	.word	0x00000020
-			ac: R_ARM_REL32	.rodata.__FUNCTION__.9433
+			ac: R_ARM_REL32	.rodata.__FUNCTION__.9392
   b0:	00000000 	.word	0x00000000
 			b0: R_ARM_GOT32	stderr
   b4:	00000018 	.word	0x00000018
-			b4: R_ARM_REL32	.LC6
+			b4: R_ARM_REL32	.LC7
 
 Disassembly of section .text.ive_obj_free_track_node_by_node:
 
@@ -658,7 +714,7 @@
   ac:	e59f2034 	ldr	r2, [pc, #52]	; e8 <ive_obj_free_track_node_by_node+0xe8>
   b0:	e1a0c001 	mov	ip, r1
   b4:	e59f1030 	ldr	r1, [pc, #48]	; ec <ive_obj_free_track_node_by_node+0xec>
-  b8:	e3003125 	movw	r3, #293	; 0x125
+  b8:	e300312f 	movw	r3, #303	; 0x12f
   bc:	e08f2002 	add	r2, pc, r2
   c0:	e7940001 	ldr	r0, [r4, r1]
   c4:	e59f1024 	ldr	r1, [pc, #36]	; f0 <ive_obj_free_track_node_by_node+0xf0>
@@ -673,8 +729,8 @@
   e4:	000000b4 	.word	0x000000b4
 			e4: R_ARM_GOTPC	_GLOBAL_OFFSET_TABLE_
   e8:	00000024 	.word	0x00000024
-			e8: R_ARM_REL32	.rodata.__FUNCTION__.9445
+			e8: R_ARM_REL32	.rodata.__FUNCTION__.9404
   ec:	00000000 	.word	0x00000000
 			ec: R_ARM_GOT32	stderr
   f0:	0000001c 	.word	0x0000001c
-			f0: R_ARM_REL32	.LC7
+			f0: R_ARM_REL32	.LC8
