--- sdk/vendor-disasm/v2_0_2_0/libmpi/mpi_region.S	2026-03-29 07:34:58.445305099 +0300
+++ sdk/vendor-disasm/v2_0_2_1/libmpi/mpi_region.S	2026-03-29 07:34:59.345307554 +0300
@@ -38,12 +38,12 @@
 00000000 l    d  .data.g_rgn_fd	00000000 .data.g_rgn_fd
 00000000 l     O .data.g_rgn_fd	00000004 g_rgn_fd
 00000000 l    d  .rodata.HI_MPI_RGN_BatchBegin.str1.4	00000000 .rodata.HI_MPI_RGN_BatchBegin.str1.4
-00000000 l    d  .rodata.__FUNCTION__.9774	00000000 .rodata.__FUNCTION__.9774
-00000000 l     O .rodata.__FUNCTION__.9774	00000013 __FUNCTION__.9774
-00000000 l    d  .rodata.__FUNCTION__.9781	00000000 .rodata.__FUNCTION__.9781
-00000000 l     O .rodata.__FUNCTION__.9781	00000015 __FUNCTION__.9781
-00000000 l    d  .rodata.__FUNCTION__.9785	00000000 .rodata.__FUNCTION__.9785
-00000000 l     O .rodata.__FUNCTION__.9785	00000018 __FUNCTION__.9785
+00000000 l    d  .rodata.__FUNCTION__.9764	00000000 .rodata.__FUNCTION__.9764
+00000000 l     O .rodata.__FUNCTION__.9764	00000013 __FUNCTION__.9764
+00000000 l    d  .rodata.__FUNCTION__.9771	00000000 .rodata.__FUNCTION__.9771
+00000000 l     O .rodata.__FUNCTION__.9771	00000015 __FUNCTION__.9771
+00000000 l    d  .rodata.__FUNCTION__.9775	00000000 .rodata.__FUNCTION__.9775
+00000000 l     O .rodata.__FUNCTION__.9775	00000018 __FUNCTION__.9775
 00000000 l    d  .rodata.mpi_rgn_check_handle.part.0.str1.4	00000000 .rodata.mpi_rgn_check_handle.part.0.str1.4
 00000000 l    d  .rodata.mpi_rgn_check_null.part.1.str1.4	00000000 .rodata.mpi_rgn_check_null.part.1.str1.4
 00000000 l    d  .rodata.mpi_rgn_open.str1.4	00000000 .rodata.mpi_rgn_open.str1.4
@@ -109,7 +109,7 @@
   50:	00000000 	.word	0x00000000
 			50: R_ARM_GOT32	stderr
   54:	0000001c 	.word	0x0000001c
-			54: R_ARM_REL32	.rodata.__FUNCTION__.9781
+			54: R_ARM_REL32	.rodata.__FUNCTION__.9771
   58:	0000001c 	.word	0x0000001c
 			58: R_ARM_REL32	.LC0
 
@@ -137,7 +137,7 @@
   40:	00000000 	.word	0x00000000
 			40: R_ARM_GOT32	stderr
   44:	0000001c 	.word	0x0000001c
-			44: R_ARM_REL32	.rodata.__FUNCTION__.9774
+			44: R_ARM_REL32	.rodata.__FUNCTION__.9764
   48:	0000001c 	.word	0x0000001c
 			48: R_ARM_REL32	.LC1
 
@@ -1323,7 +1323,7 @@
  134:	00000000 	.word	0x00000000
 			134: R_ARM_GOT32	stderr
  138:	00000064 	.word	0x00000064
-			138: R_ARM_REL32	.rodata.__FUNCTION__.9785
+			138: R_ARM_REL32	.rodata.__FUNCTION__.9775
  13c:	00000064 	.word	0x00000064
 			13c: R_ARM_REL32	.LC4
 
